src/analysis/processing/qgsalgorithmreclassifybylayer.cpp

```diff
@@ -15,6 +15,7 @@
  *                                                                         *
  ***************************************************************************/
 
+#include "gdal.h"
```

Comment:
Question: what is most efficient, qgsgdalutils or gdal (they both work)? I browsed some processing files and couldn't find a common practice (even found places they were both included)

Reply:
qgdgdalutils.h does "#include" in turn gdal.h.

Anyway I think `GDALVersionInfo("VERSION_NUM")` should be used instead of `GDAL_VERSION_NUM`: the former is the GDAL version actually used by QGIS at runtime, while the latter is the GDAL version used at compile time.

`GDALVersionInfo("VERSION_NUM")` is the same as `GdalUtils.version()` used for the Python processing algorithms.